UMN-ActiveDirectory/Public/Confirm-ADObjectDCReplication.ps1

```diff
@@ -0,0 +1,75 @@
+﻿<#
+		.SYNOPSIS
+		Cmdlet that returns true if the object exists on all domain controllers within the domain and false if it doesn't.
+		
+		.DESCRIPTION
+		A cmdlet that take in the identity of an object, an object type and an optional maximum wait time (default is 30 seconds)
+        to determine if that object exists on all domain controllers within a domain. The script tries once each second up until
+        thte maximum wait time is reached before returning a false if not all domain controllers return true.
+		
+		.PARAMETER ADObject
+		A string representing the object, can be a DN, GUID, sAMAccountName or SID.
+		
+		.PARAMETER Type
+		The type of object being tested.  The unknwon identity simply uses Get-ADObject rather than an object specific locator.
+		
+		.PARAMETER MaxWait
+		The maximum time to wait in seconds before determining that the object does not exist on all domain controllers.
+		
+		.EXAMPLE
+		Confirm-ADObjectDCReplication -ADObject "FooBar" -Type "Computer" -MaxWait 5
+		
+		.EXAMPLE
+		Confirm-ADObjectDCReplication -ADObject "foobar" -Type "User"
+		
+		.EXAMPLE
+		Confirm-ADObjectDCReplication -ADObject "S-1-5-1-5125-16836816-12512325" -Type "Group"
+		
+		.EXAMPLE
+		Confirm-ADObjectDCReplication -ADObject "CN=Foo,CN=Bar,DC=domain,DC=acme,DC=com" -Type "OU"
+	#>
+function Confirm-ADObjectDCReplication {
+	[CmdletBinding()]
+	param(
+		[Parameter(Mandatory=$true)][string]$ADObject,
+		[Parameter(Mandatory=$true)][ValidateSet("Computer", "User", "Group", "OU", "Unknown")][string]$Type,
+		[Parameter(Mandatory=$false)][int]$MaxWait=30
+	)
+
+	$DomainControllers = New-Object System.Collections.ArrayList
+	$GoodServers = New-Object System.Collections.ArrayList
+
+	$Count = 1
+
+	try {
+		# Get the list of domain controllers to check
+		$null = Get-ADDomainController -Filter * | Foreach-Object { $DomainControllers.Add($_.HostName) }
```

Comment:
this works but is kinda convoluted...  Not sure if this is actually better but it would also work and maybe be clearer?
`$DomainControllers.AddRange(@((Get-ADDomainController -filter *).HostName))`

No that sucks too.  Maybe brevity is not a good idea here.
```
$DomainControllersToCheck = New-Object System.Collections.ArrayList
$DomainControllers = (Get-ADDomainController -Filter *).HostName
$DomainControllersToCheck.AddRange(@($DomainControllers))
```
You can add multiple objects to an arraylist using the addrange method but it has to be a container, so wrapping it in `@()` ensures that even if there's only one DC it will still.  So I don't actually know if that's better.

Reply:
Meh, this works for now.  Once we start leveraging this more in production for some of the file share automation project I'll come back and rework.